### DISCUSSION TITLE
Traffic Stats - Reset with WorkManager

### DISCUSSION THREAD

**Discussion by w3rkzeug** [schwabe/ics-openvpn#1295] [Q&A]
Hi there, 
I wonder if anyone tried collecting the bandwidth usage for wifi/mobile data not being session dependant ?
Like for example, setting up a 1hour periodic work with WorkManager, that will create a datapoint, get the current usage & send to 3rd party api, next hour the worker will reset the counter & repeat the procedure.
This model should work, basicly the question is How do i reset the traffic counters while the client is CONNECTED ?

**Comment by w3rkzeug** [comment]
Solved, Collecting the diffIn , diffOut on a prefferred schedule did the job.
Thanks

**Comment by w3rkzeug** [comment]
Discrepencies occured, digged into deeper , looks like these values depend on TrafficHistory datapoint, which is created during the start of the VPN Service...Any thoughts how can this be accomplished ?
thanks

**Comment by schwabe** [comment]
I am not understanding the problem you are trying to solve. WorkManager is an Android service. I am not sure how that relates to my app.

**Reply by w3rkzeug** [reply]
Sorry maybe i wasn't too clear , 
So im using your app's api , i need to get the hourly bandwidth usage, in my understanding i need WorkManager to have a work that will run each hour, get the bandwidth usage, which will further send this data to an api endpoint.

**Reply by schwabe** [reply]
Ah I understand now. I never really designed the API for this usecase. But you might also look into extending the API to export more of the internal TrafficStatistics data point than what is available now. (PRs are welcome).

**Reply by w3rkzeug** [reply]
Do you have any suggestion to a better approach to extend the API for this ?
